packages/eslint-plugin-prairielearn/src/rules/safe-db-types.ts

```diff
@@ -0,0 +1,592 @@
+                      messageId: 'unsafeTypes',
+                      data: { propName, typeName },
+                    });
+                  }
+                }
+              }
+            }
+          } else if (ts.isTypeAliasDeclaration(decl) && decl.type) {
+            // Type alias might be an inline object type
+            if (ts.isTypeLiteralNode(decl.type)) {
+              for (const member of decl.type.members) {
+                if (ts.isPropertySignature(member) && member.type && member.name) {
+                  if (!ts.isIdentifier(member.name)) continue;
+
+                  const propName = member.name.text;
+                  const violations = checkTypeNodeForDbTypes(member.type, typeChecker);
+
+                  if (violations.length > 0) {
+                    // Find the JSX attribute for this prop
+                    const attribute = childOpeningElement.attributes.find(
+                      (attr) =>
+                        attr.type === 'JSXAttribute' &&
+                        attr.name.type === 'JSXIdentifier' &&
+                        attr.name.name === propName,
+                    );
+
+                    for (const typeName of violations) {
+                      if (isTypeInAllowlist(typeName, allowlist)) continue;
+
+                      context.report({
+                        node: attribute || reportNode,
+                        messageId: 'unsafeTypes',
+                        data: { propName, typeName },
+                      });
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // Check for spread attributes
+  const attributes = childOpeningElement.attributes;
+  for (const attr of attributes) {
+    if (attr.type === 'JSXSpreadAttribute') {
+      context.report({
+        node: reportNode,
+        messageId: 'spreadAttributes',
+      });
+      continue;
+    }
+  }
+}
+
+export default ESLintUtils.RuleCreator.withoutDocs<
+  [{ allowDbTypes?: (string | RegExp)[] }?],
+  'spreadAttributes' | 'unsafeTypes'
+>({
+  meta: {
+    type: 'problem',
+    messages: {
+      spreadAttributes: 'Spread attributes are not allowed in Hydrate children.',
+      unsafeTypes:
+        'Prop "{{propName}}" uses type "{{typeName}}" which is derived from db-types.ts. Use safe-db-types.ts instead.',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowDbTypes: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [{}],
+  create(context) {
+    const options = context.options[0] || {};
+    const allowlist = options.allowDbTypes || [];
+    return {
+      JSXElement(node) {
+        const openingElementNameExpression = node.openingElement.name;
+
+        // TODO: handle namespace and member expressions?
```

Comment:
Intentional?

Reply:
I think this isn't needed. Comment removed in 58f6a5b.